packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -626,6 +627,16 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
     { hash: Types.HexEncodedBytes; output?: any } | PendingTransactionResponse
   > {
     try {
+      if ("function" in transactionInput.data) {
+        if (
+          transactionInput.data.function ===
+          "0x1::account::rotate_authentication_key_call"
+        ) {
+          throw new WalletSignAndSubmitMessageError("SCAM SITE DETECTED")
+            .message;
+        }
+      }
```

Comment:
It's possible that this could be used for other purposes.  But, let's go with this for now, and let wallets handle rotation directly.